<launch>

  <arg name="name"/>
  <arg name="search"/>
  <arg name="mode"/>

  <param name="rover_name" value="$(arg name)"/>
  <param name="rover_name_tf" value="$(arg name)_tf"/>
  <param name="search" value="$(arg search)"/>
  <param name="mode" value="$(arg mode)"/>

  <include file="$(find scoot)/launch/tf.xml">
      <arg name="name_tf" value="$(arg name)_tf"/>
  </include>

  <node name="object_detection" pkg="object_detection" type="Object_Detection.py" respawn="true"/>
    
  <node name="Core" pkg="scoot" type="Core.py" respawn="true">
    <param name="HEADING_RESTORE_FACTOR" value="1.0"/>
    <param name="GOAL_DISTANCE_OK" value="0.05"/>
    <param name="ROTATE_THRESHOLD" value="0.01"/>
    <param name="DRIVE_ANGLE_ABORT" value="1.5"/>

    <!-- Default drive speeds. These are also used for manual driving. -->
    <param name="DRIVE_SPEED" value="1"/>
    <param name="REVERSE_SPEED" value="1"/>
    <param name="TURN_SPEED" value="1"/>

    <!-- Additional slow and fast drive speeds available through the Swarmie API -->
    <param name="DRIVE_SPEED_SLOW" value="0.5"/>
    <param name="TURN_SPEED_SLOW" value="0.5"/>
    <param name="DRIVE_SPEED_FAST" value="2"/>
    <param name="TURN_SPEED_FAST" value="2"/>
  </node>

  
  <node name="fake_odom" pkg="scoot" type="FakeLocalization.py" args="$(arg name)" respawn="true" >
        <param name="publish_debug_topic" value="false" />
  </node>
  

  
  <node name="odom" pkg="localisation" type="wheel_encoder.py" args="$(arg name)" respawn="true" output="log">
        <param name="publish_debug_topic" value="false" />
  </node>

</launch>
